esinstall/src/entrypoints.ts

```diff
@@ -218,14 +214,15 @@ export function resolveEntrypoint(
     packageLookupFields,
   });
 
-  // Some packages are types-only. If this is one of those packages, resolve with that.
-  if (!foundEntrypoint && hasTypes(depManifest)) {
-    const typesLoc = (depManifest.types || depManifest.typings) as string;
-    return path.join(depManifestLoc, '..', typesLoc);
-  }
   // Sometimes packages don't give an entrypoint, assuming you'll fall back to "index.js".
   if (!foundEntrypoint) {
-    foundEntrypoint = 'index.js';
+    for(let possibleEntrypoint of ['index.js', 'index.json', depManifest.types || depManifest.typings || 'index.d.ts']) {
```

Comment:
this still shouldn't be returning a types file though. If I'm trying to install a package like `csstypes` with esinstall, I'd expect it to either fail loudly or silently, but it shouldn't install a `css-types/index.d.ts` file in my all-JavaScript web_modules folder.  

 +1 for checking for and resolving `index.js` and `index.json` though.

Reply:
This doesn't result in an index.d.ts being put into the web_modules folder. I guess somewhere else in esinstall omits the file.

We could maybe make this return `null` but that would be a breaking change to the function (do we care about that)?

Reply:
DTS is ignored here: https://github.com/snowpackjs/snowpack/blob/a7f19172b542340fdba367802903d54e7b3d848d/esinstall/src/index.ts#L285-L288

Seems like we do expect to return the type file here as there are other references to `type: 'DTS'`